NotificationSystem/NotificationSystem/Models/OrcasiteHelper.cs

```
@@ -319,5 +355,130 @@ public async Task<bool> PostDetectionAsync(string json)
+        /// requires querying S3 to find the .ts folders.  Hence, this should only be
+        /// done when it is known that the timestamps are incorrect.
+        /// </summary>
+        /// <param name="originalInput">List of detections to process</param>
+        /// <returns>List of corrected detections</returns>
+        public async Task<List<JsonElement>> FixTimestampsAsync(IReadOnlyList<JsonElement> originalInput)
+        {
+            var correctedInput = new List<JsonElement>();
+
+            foreach (JsonElement originalDetection in originalInput)
+            {
+                DateTime? originalDateTime = TryGetTimestamp(originalDetection);
+                if (originalDateTime == null)
+                {
+                    continue;
+                }
+
+                // Convert dateTime to Unix time (seconds since 1970-01-01).
+                var originalDateTimeOffset = new DateTimeOffset(originalDateTime.Value);
+                long originalUnixTimeSeconds = originalDateTimeOffset.ToUnixTimeSeconds();
+
+                // Fix the Unix time.  The originalDateTime is incorrect and
+                // was computed based on clips being 11 seconds long instead
+                // of 10 seconds long.  We can correct this once we know the
+                // .ts folder date to start calculating the drift based on.
+
+                string locationIdString = TryGetLocationIdString(originalDetection);
+                if (string.IsNullOrEmpty(locationIdString))
+                {
+                    continue;
+                }
+
+                List<string> folders = await GetPublicS3FoldersAsync(locationIdString);
+
+                // Find the most recent folder older than originalUnixTimeSeconds.
+                long folderTimeSeconds = 0;
+                foreach (var folderName in folders)
+                {
+                    if (long.TryParse(folderName, out long unixTime))
```

What is the desired behavior if the parse fails? Should we error out instead of silently skipping?